controllers/servicefence_controller.go

```diff
@@ -376,17 +375,61 @@ func addDomainsWithHost(domains map[string]*lazyloadv1alpha1.Destinations, sf *l
 
 	for h, strategy := range sf.Spec.Host {
 
+		// handle namespace level host, like 'default/*'
+		if strings.HasSuffix(h, "/*") {
+			subHosts := strings.Split(h, "/")
+			if len(subHosts) != 2 {
+				log.Errorf("%s is invalid host, skip", h)
+				continue
+			}
+			nsSvcCache.RLock()
+			svcs := nsSvcCache.Data[subHosts[0]]
+			var allHost []string
+			for svc := range svcs {
+				subSvcs := strings.Split(svc, "/")
+				service := fmt.Sprintf("%s.%s.svc.cluster.local", subSvcs[1], subSvcs[0])
+				if !isValidHost(service) {
+					continue
+				}
+				if domains[service] != nil {
+					continue
+				}
+				// service relates to other services
+				if destSvcs := getDestination(service); len(destSvcs) > 0 {
+					// ignore destSvc that in the same namespace
+					for i := 0; i < len(destSvcs); {
+						subDestSvc := strings.Split(destSvcs[i], ".")
+						if subDestSvc[1] == subHosts[0] {
+							destSvcs[i], destSvcs[len(destSvcs)-1] = destSvcs[len(destSvcs)-1], destSvcs[i]
+							destSvcs = destSvcs[:len(destSvcs)-1]
+						} else {
+							i++
+						}
+					}
+
+					allHost = append(allHost, destSvcs...)
+				}
+			}
+			domains[h] = &lazyloadv1alpha1.Destinations{
+				Hosts:  allHost,
+				Status: lazyloadv1alpha1.Destinations_ACTIVE,
+			}
+			nsSvcCache.RUnlock()
+			continue
+		}
+
+		// handle service level host, like 'a.default.svc.cluster.local' or 'a' or 'a.default'
 		fullHost := h
-		subDomains := strings.Split(h, ".")
-		switch len(subDomains) {
+		subHosts := strings.Split(h, ".")
+		switch len(subHosts) {
 		// full service name, like "reviews.default.svc.cluster.local", needs no action
 		case 5:
 		// short service name without namespace, like "reviews", needs to add namespace of servicefence and "svc.cluster.local"
 		case 1:
-			fullHost = fmt.Sprintf("%s.%s.svc.cluster.local", subDomains[0], sf.Namespace)
+			fullHost = fmt.Sprintf("%s.%s.svc.cluster.local", subHosts[0], sf.Namespace)
 		// short service name with namespace, like "reviews.default", needs to add "svc.cluster.local"
 		case 2:
-			fullHost = fmt.Sprintf("%s.%s.svc.cluster.local", subDomains[0], subDomains[1])
+			fullHost = fmt.Sprintf("%s.%s.svc.cluster.local", subHosts[0], subHosts[1])
 		default:
 			log.Errorf("%s is invalid host, skip", h)
```

Comment:
suppport it now or later? maybe hard to distinguish from `svc.ns` format.

Reply:
will update these check logic when adapting serviceEntry in future